be/src/storage/vectorized/schema_change.cpp

```diff
@@ -641,7 +641,7 @@ bool ChunkMerger::_pop_heap() {
     return true;
 }
 
-bool LinkedSchemaChange::process(vectorized::TabletReader* reader, RowsetWriter* new_rowset_writer,
+bool LinkedSchemaChange::process(std::shared_ptr<vectorized::TabletReader> reader, RowsetWriter* new_rowset_writer,
```

Comment:
pass a shared_ptr is have a high overhead. You can still use pointer pass the parameter.